services/editor/src/pages/settings/components/HooksPage/HookTypes.js

```diff
@@ -1,4 +1,7 @@
-export const hookTypes = [{ label: 'Notification Webhook', value: 'notification_webhook' }];
+export const hookTypes = [
+  { label: 'Notification Webhook', value: 'notification_webhook' },
+  { label: 'Slack Webhook', value: 'slack_webhook' },
```

Comment:
Can't we create the slack as a subtype of the notification webhook?

Reply:
with something like format/template field

Reply:
The code is not really classes and generics, It's just a switch case and some functions, adding format would reuqire to add a template engine which is too much for me now

Reply:
no need, even `format: slack` works, just that I'll be a subtype of notification webhook I think

Reply:
Since the difference should be just in the format

Reply:
That's different from hook that is integrated with slack API, or email hook, or validation/mutation webhook, etc...

Reply:
for example:
```
      var response = await _client.PostAsync(url, new StringContent( JsonConvert.SerializeObject(new { text }), Encoding.UTF8, "application/json"));
      response.EnsureSuccessStatusCode();
    }
```
There shouldn't be duplication of response sending as only the payload is different

Reply:
but the format can be static at this point (simple function) without any template engine, and in the future we can think of extending it

Reply:
It also affects the unit test as it's only need to check for post data and not a side effect for each format.

Reply:
How will I choose the format in the UI? it means that I need another field, right?

Reply:
Yes, but if we had tags you'll need to touch the UI either way :)

Reply:
you got me there